### ISSUE TITLE
Directus fails to start with named pipes

### ISSUE THREAD

**Issue by nichols-green** [directus/directus#5977]
Directus fails to start when using a named pipe (i.e. `\\.\pipe\pipe-name`) in place of port. This is necessary for iisnode, and azure, where iisnode is utilized. iisnode sets the `PORT` environment variable to a named pipe in order to dynamically bind sites to the app pool and server IP. 

The express, http and net modules all support support named pipes in addition to the more typical numbered ports, yet the [.env module](https://github.com/directus/directus/blob/main/api/src/env.ts) restricts the port variable to a number. I did some testing by setting the type to string, and directus was able start and listen for requests with both a numbered port, and a named pipe. 

~The resulting req shape when using a named pipe is slightly different, and this may require some additional refactoring, but~ IMO this is necessary to support iis and azure users. 

*EDIT* The req shape isn't different, this was related to a parallel issue of setting the proper configuration in the web.config.

**Comment by nichols-green** [comment]
For anyone working on this or exploring this, iisnode acts as a reverse proxy, and won't send x-forwarded headers by default. These headers are necessary for the authentication module and likely others. In order to enable x-forwarded headers add `<iisnode enableXFF="true" />` to the web.config.

**Comment by rijkvanzanten** [comment]
@nichols-green Just so I understand correctly (i'm not familiar with the `\\.\pipe\pipe-name` syntax), is it just a drop-in replacement for the numbered port?

**Comment by nichols-green** [comment]
> @nichols-green Just so I understand correctly (I'm not familiar with the `\\.\pipe\pipe-name` syntax), is it just a drop-in replacement for the numbered port?

@rijkvanzanten while I'm not an expert in this either, that is my understanding. The iisnode documentation dictates that the server must listen to `process.env.PORT` as iisnode sets that environment variable dynamically. `server.listen` is overloaded to accept a port, or a pipe, and the rest is abstracted [source](https://nodejs.org/api/net.html#net_server_listen). 

To test this, I deployed my directus site to the server, then remoted into the server and manually altered the `dist/env.js` file changing `PORT: 'number' to `PORT: 'string'. After an iis reset, I was able to launch the site and login with the admin credentials without error. 

I also ran this locally with the same change where I'm not using iis. In this scenario the PORT variable is set to a numbered port, and again, the server started without error. 

I started a PR with this one-line change https://github.com/directus/directus/pull/6001
